src/object/srv_obj.c

```diff
@@ -2075,7 +2075,7 @@ ds_obj_punch_handler(crt_rpc_t *rpc)
 	exec_arg.flags = flags;
 	/* Execute the operation on all shards */
 	rc = dtx_leader_exec_ops(&dlh, obj_tgt_punch, &exec_arg);
-out:
```

Comment:
Directly move label "out" from line 2078 to line 2096, then do not need new label "dtx_end".